emmarin/cl/ledger/src/partial_tx.rs

```diff
@@ -0,0 +1,82 @@
+use ledger_proof_statements::ptx::{PtxPrivate, PtxPublic};
+
+use crate::error::{Error, Result};
+use cl::cl::{merkle, PartialTx, PartialTxWitness};
+use cl::zone_layer::notes::ZoneId;
+
+#[derive(Debug, Clone)]
+pub struct ProvedPartialTx {
+    pub ptx: PartialTx,
+    pub cm_root: [u8; 32],
+    pub risc0_receipt: risc0_zkvm::Receipt,
+}
+
+impl ProvedPartialTx {
+    pub fn prove(
+        ptx_witness: PartialTxWitness,
+        input_cm_paths: Vec<Vec<merkle::PathNode>>,
+        cm_root: [u8; 32],
+        from: Vec<ZoneId>,
+        to: Vec<ZoneId>,
```

Comment:
We can derive this from the ptx_witness now

Reply:
mmm no? ptx_witness do not have info about the zones unless we want to change that

Reply:
It's in the outputs no? `to == [o.zone_id for o in ptx_witness.outputs]`

Reply:
ah got it, yeah lets add it to the witness'

Reply:
fixed!